api/runnerpool/placer_tracker.go

```diff
@@ -114,15 +114,13 @@ func (tr *placerTracker) HandleDone() {
 // RetryAllBackoff blocks until it is time to try the runner list again. Returns
 // false if the placer should stop trying.
 func (tr *placerTracker) RetryAllBackoff(numOfRunners int, err error) bool {
-
-	// If the last call to obtain runners failed due to a user error or
-	// misconfiguration then do not re-attempt to obtain a runner.
-	if err != nil {
-		tr.HandleFindRunnersFailure(err)
+	// If there are no runners and last call to provision runners failed due
+	// to a user error (or misconfiguration) then fail fast.
+	if numOfRunners == 0 && err != nil {
```

Comment:
Not sure I agree with this check on the number of runners.
Adding this means that the user will be notified of the misconfiguration only if she has not runners or, in case has some old runners, only when they get recycled. 
I'd prefer to immediately notify the users about the error so they can fix it soon. All the point of this change is to fail fast and give a quick feedback to customers, if we check for empty runner list we miss the point of this change.
Let's consider this scenario where I make a change (misconfiguring my env) and deploy it, I test that it works and i managed to invoke a function because I land on an old runner, after "n" hours my functions suddenly start to fail because the old good runners get recycled and the new ones are affected by my misconfiguration. Personally I prefer to find the error soon so i have the right context at the right moment to fix it. What do you think?

Reply:
in any case I think that we still record the stats for numOfRunners==0

Reply:
Yes, I agree. It is a choice between two behaviors, but, my gut feeling is I prefer fast feedback when I am/have done something wrong. I can revert the check for existing runners if we all agree this is most useful/suitable.

Reply:
I like the tradeoff of inspecting the error and fail fast if the error is a policy misconfiguration, that is a specific error for a specific service implementation, not sure how to make it abstract here, maybe some sort of pluggable error handling ? just a random idea not sure the effort to implement that.

Reply:
if it is too difficult or taking too long to implement the inspect of errors I think we can go with this solution as it is now with the numOfRunners check and having a follow up patch.

Reply:
@skinowski  - I was going to use the actual error, but, it was just a string specific to the OCI backend implementation I was using, I assumed we would want a more detailed abstraction here and not leak what is underneath (OCI). The IsFuncError() is what (to me) looked like the current way of determining a 'user blame' error? I also would prefer something better and clearer than using IsFuncError, but, that should not be a unilateral. Wanna help?

I agree with @ostrain . I currently think we should only skip when there are no runners. Looks like lots of intermittent things could go wrong inside the GetRunners call e.g. periodic shonky DNS.

Reply:
To add my $0.02

I think trying to extend this to cover cases where runner count is already greater than 0 may be a good move long term but will introduce complications.
In the short term handling the 0 runners case effectively is a good basis on which to improve.

Using the function error interface seems like the right way to determine blame, or, if we want to consider this tangential to invocation errors we could introduce a new interface representing user blame errors in the runner pool implementation.

Reply:
What actionable changes do I need to make, or, further tests to add/perform? Thanks!

Can I confirm IsFunc() is the mandated way to determine this is a user blame error?

Reply:
This works as written.